apps/src/code-studio/assets/DefaultSpritesEditor.jsx

```diff
@@ -38,23 +35,65 @@ export default class DefaultSpritesEditor extends React.Component {
   };
 
   deleteSpriteFromDefaults = spriteName => {
-    let updatedList = {...this.state.defaultList};
-    delete updatedList[spriteName];
+    let updatedList = [...this.state.defaultList];
+    // Find the index at which the sprite should be deleted
+    for (let index = 0; index < updatedList.length; index++) {
+      let sprite = updatedList[index];
+      if (sprite.name === spriteName) {
+        // Item is to be deleted
+        updatedList.splice(index, 1);
+      }
+    }
     this.setState({defaultList: updatedList});
     this.incrementPendingChanges();
   };
 
-  addSpriteToDefaults = (spriteName, spriteCategory) => {
-    let updatedList = {...this.state.defaultList};
-    updatedList[spriteName] = {name: spriteName, key: spriteCategory};
+  addSpriteToDefaults = (addToBeginning, spriteName, spriteCategory) => {
+    let updatedList = [...this.state.defaultList];
+    if (addToBeginning) {
+      updatedList.unshift({name: spriteName, key: spriteCategory});
+    } else {
+      updatedList.push({name: spriteName, key: spriteCategory});
+    }
     this.setState({defaultList: updatedList});
     this.incrementPendingChanges();
   };
 
+  reorderSpriteByOne = (moveForward, spriteName) => {
+    let updatedList = [...this.state.defaultList];
+    let originalIndex = -1;
+    // Find index
+    for (let index = 0; index < updatedList.length; index++) {
+      if (updatedList[index].name === spriteName) {
+        originalIndex = index;
+      }
+    }
+
+    // If the original index is less than 0, take no action
+    if (originalIndex >= 0) {
```

Comment:
nit: i think it's clearer to return early in this case, and it reduces nesting, which i think helps readability

```javascript
if (originalIndex < 0) {
  return;
}
```